### DISCUSSION TITLE
Best practices should have way more "do's" than "don'ts"

### DISCUSSION THREAD

**Discussion by rannyh** [lfn-cnti/bestpractices#33] [General]
It would be highly valuable if the output of this workgroup will focus on filling the gap between the set of principles/factors/guidelines, and the conformance test suite that validates adherence to the principles. The principles and factors tend to have a lot of "don'ts". We can provide value by adding the "do's" that will help application developers make sure they will successfully pass the conformance test. A good reference example in my view could be something like this guide from GCP:
https://cloud.google.com/solutions/twelve-factor-app-development-on-gcp
that points developers to specific practices on GCP that will enable them to build compliant 12 factor apps (https://12factor.net)

**Comment by electrocucaracha** [comment]
Or at least more "do's" than "don'ts"
